src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -247,6 +243,13 @@ public SettableFuture<Long> load(Digest digest) {
   private final transient Entry header = new SentinelEntry();
   private volatile long unreferencedEntryCount = 0;
 
+  private static final int MAX_ENTRIES = 10000;
+  private static final Cache<String, DigestFunction.Value> digest_to_function_map =
+          Caffeine.newBuilder()
+                  .expireAfterWrite(2, TimeUnit.MINUTES)
```

Comment:
What happens if there are more than 10k write requests in 2 mins.

Reply:
Doesn't matter. This is basically a LRU cache. As long as there are no 10k request (or 2 minutes) between the methods `getOutput` and `putOrReferenceGuarded` it works. But we can also increase the size if you think that is not sufficient.

Reply:
You may increase it, but doing so will add extra overhead. Using an additional data structure (accessible by multiple threads) to pass a value from one method to another may seem like a shortcut to avoid further code changes. However, it's better to make the necessary changes rather than introduce a new data structure.

Reply:
Fair, but it requires quite a large code change then, even more if we don't want to create a new overload method like you mentioned but change the existing interface methods as well as all implementations of those

Reply:
Yes, there will be small changes at multiple places in the code. Lets try to do it proper way so it can be upstreamed.